models/ldndc/R/write.config.LDNDC.R

```diff
@@ -0,0 +1,202 @@
+  # Create launch script (which will create symlink)
+  if (!is.null(settings$model$jobtemplate) && file.exists(settings$model$jobtemplate)) {
+    jobsh <- readLines(con = settings$model$jobtemplate, n = -1)
+  } else {
+    jobsh <- readLines(con = system.file("template.job", package = "PEcAn.LDNDC"), n = -1)
+  }
+  
+  # Create host specific settings
+  # NOT MEANINGFUL HERE -----
+  hostsetup <- ""
+  if (!is.null(settings$model$prerun)) {
+    hostsetup <- paste(hostsetup, sep = "\n", paste(settings$model$prerun, collapse = "\n"))
+  }
+  if (!is.null(settings$host$prerun)) {
+    hostsetup <- paste(hostsetup, sep = "\n", paste(settings$host$prerun, collapse = "\n"))
+  }
+  
+  hostteardown <- ""
+  if (!is.null(settings$model$postrun)) {
+    hostteardown <- paste(hostteardown, sep = "\n", paste(settings$model$postrun, collapse = "\n"))
+  }
+  if (!is.null(settings$host$postrun)) {
+    hostteardown <- paste(hostteardown, sep = "\n", paste(settings$host$postrun, collapse = "\n"))
+  }
+  
+  #------
+  
+  
+  # Create job.sh based on the given settings
+  jobsh <- gsub("@HOST_SETUP@", hostsetup, jobsh)
+  jobsh <- gsub("@HOST_TEARDOWN@", hostteardown, jobsh)
+  
+  jobsh <- gsub("@SITE_LAT@", settings$run$site$lat, jobsh)
+  jobsh <- gsub("@SITE_LON@", settings$run$site$lon, jobsh)
+  
+  jobsh <- gsub("@START_DATE@", settings$run$start.date, jobsh)
+  jobsh <- gsub("@END_DATE@", settings$run$end.date, jobsh)
+  
+  jobsh <- gsub("@OUTDIR@", outdir, jobsh)
+  jobsh <- gsub("@RUNDIR@", rundir, jobsh)
+  jobsh <- gsub("@METPATH@", MetPath, jobsh)
+  
+  jobsh <- gsub("@BINARY@", paste(settings$model$binary, paste0(rundir, "/project.ldndc")), jobsh)
+  
+  # if(is.null(settings$model$delete.raw)){
+  #   settings$model$delete.raw <- FALSE
+  # }
+  # 
+  # jobsh <- gsub("@DELETE.RAW@", settings$model$delete.raw, jobsh)
+  
+  # Write job.sh file to rundir
+  writeLines(jobsh, con = file.path(settings$rundir, run.id, "job.sh"))
+  Sys.chmod(file.path(settings$rundir, run.id, "job.sh")) # Permissions
+  
+  
+  
+  #### write run-specific PFT parameters here #### Get parameters being handled by PEcAn
+  speciesparfile <- readLines(con = system.file("speciesparameter_template.xml", package = "PEcAn.LDNDC"), n = -1)
+  
+  
+  #----------------------
+    
+  ## Set-up the necessary files in to the run directory so
+  ## model is able to function properly. Later on, these
+  ## files should be populated with initial values.
+    
+  
+  ###### THIS NEEDS TO BE FUNCTION AT SOME POINT
+  ### PROBABLY SIMILAR FUNCTION FOR siteparameters as well
+  #
+  mnemonic_1 <- "__grasses__"
+  group <- "grass"
+  mnemonic_2 <- "perennialgrass"
+  
+  a.1 <- paste0("<species mnemonic='", mnemonic_1, "' group='", group, "' > \n")
+  b.1 <- paste0("\t\t\t\t\t<species mnemonic='", mnemonic_2, "' > \n")
+  b.2 <- apply(trait.values[[1]], 1, function(x){paste0("\t\t\t\t\t\t<par name='", names(x), "' value='", x, "' /> \n")})
```

Comment:
not sure but this implementation could cause problems later on L162 and L164 if more than one trait values passed by PEcAn (then you'll have a vector of two or more to gsub `Info` with). But more importantly it assumes pecan-bety names match LDNDC parameter names (which was the case for `SLA`). How about starting a small backbone of a typical PEcAn write.configs loop over trait values here, e.g. something like:

```
  b.2 <- ""
  for (pft in seq_along(trait.values)) {
    
    pft.traits <- unlist(trait.values[[pft]])
    pft.names <- names(pft.traits)
    
    if ("SLA" %in% pft.names) {
      b.2 <- paste(b.2, paste0("\t\t\t\t\t\t<par name='SLA' value='", pft.traits[which(pft.names == "SLA")], "' /> \n"), collapse="")
    }
    
    if ("extinction_coefficient_diffuse" %in% pft.names) {
      b.2 <- paste(b.2, paste0("\t\t\t\t\t\t<par name='ext' value='", pft.traits[which(pft.names == "extinction_coefficient_diffuse")], "' /> \n"), collapse="")
    }
  }
```

Reply:
Good points, changed the code accordingly. This need to be designed and rewritten later when ready to run with multiple pfts.